torchtitan/train.py

```diff
@@ -641,6 +641,9 @@ def state_dict(self) -> dict[str, Any]:
     def load_state_dict(self, state_dict: dict[str, Any]):
         self.step = state_dict["step"]
         self.ntokens_seen = state_dict["ntokens_seen"]
+        if not self.job_config.checkpoint.enable_ft and self.ft_manager is not None:
```

Comment:
Setting dataloader state_dict is not safe. data.loader_state_dict() may be invoked before or after this `load_state_dict()`.